### PR TITLE
Bugfix/atr 801 dev forbid non public graphs and dacs

### PR DESCRIPTION
**Pull request by SENya1990**
Changes overview:
- reworked PX1022 to report non public DACs and graphs
- reworked code fix
- added unit tests for non public graph and DAC
- added and integrated localization string utils
- updated documentation for PX1022

### CODE REVIEW

File: docs/Summary.md [hunk 1]
```diff
@@ -32,7 +32,7 @@ Acuminator does not perform static analysis of projects whose names contain `Tes
 | [PX1019](diagnostics/PX1019.md) | A DAC property field with the `AutoNumber` attribute must have the string type.                                  | Error | Available   |
 | [PX1020](diagnostics/PX1020.md) | For the DAC property with the `AutoNumber` attribute, the string length must be greater than or equal to PX.Objects.CS.NumberingSequence.StartNbr value.     | Error | Available   |
 | [PX1021](diagnostics/PX1021.md) | The type of the DAC field attribute does not correspond to the property type. | Error | Available |
-| [PX1022](diagnostics/PX1022.md) | The non-public graph and DAC extensions are not supported.                                                       | Error | Available   |
+| [PX1022](diagnostics/PX1022.md) | The non-public graphs, DAC, graph and DAC extensions are not supported. | Error | Available   |
```
Comment (SENya1990) [inline docs/Summary.md]: @EAndrosova please review changes in the diagnostic summary

File: docs/diagnostics/PX1022.md [hunk 2]
```diff
@@ -5,16 +5,67 @@ This document describes the PX1022 diagnostic.
 
 | Code   | Short Description                                                                                                | Type  | Code Fix    | 
 | ------ | ---------------------------------------------------------------------------------------------------------------- | ----- | ----------- | 
-| PX1022 | The non-public graph and DAC extensions are not supported.                                                       | Error | Available   |
+| PX1022 | The non-public graphs, DAC, graph and DAC extensions are not supported.                                          | Error | Available   |
```
Comment (SENya1990) [inline docs/diagnostics/PX1022.md]: @EAndrosova please review changes in the diagnostic documentation

File: src/Acuminator/Acuminator.Analyzers/DiagnosticsShortName.resx [hunk 3]
```diff
@@ -396,4 +396,10 @@
   <data name="PX1027ForbiddenCompanyPrefixInDacField" xml:space="preserve">
     <value>ForbiddenCompanyPrefixInDacField</value>
   </data>
+  <data name="PX1022Dac" xml:space="preserve">
```
Comment (SENya1990) [inline src/Acuminator/Acuminator.Analyzers/DiagnosticsShortName.resx:399]: @EAndrosova please review new short names for the diagnostic

File: src/Acuminator/Acuminator.Analyzers/Resources.resx [hunk 4]
```diff
@@ -504,17 +504,17 @@
   <data name="PX1020Fix" xml:space="preserve">
     <value>Make the string length sufficient for the auto-numbering</value>
   </data>
-  <data name="PX1022DacFix" xml:space="preserve">
+  <data name="PX1022DacExtensionFix" xml:space="preserve">
```
Comment (SENya1990) [inline src/Acuminator/Acuminator.Analyzers/Resources.resx:507]: @EAndrosova please review changes in the PX1022 diagnostic messages and code fix messages

**Review by dropsonic — APPROVED** [review]

**Review by EAndrosova — APPROVED** [review]

**Pull request merged**